src/coreclr/tools/Common/TypeSystem/Ecma/EcmaSignatureParser.cs

```diff
@@ -616,7 +617,7 @@ public MarshalAsDescriptor ParseMarshalAsDescriptor()
                         _reader.ReadSerializedString();
 
                         // Read managed marshaler name
-                        _reader.ReadSerializedString();
+                        managedMarshallerTypeName = _reader.ReadSerializedString();
 
                         // Read cookie
                         _reader.ReadSerializedString();
```

Comment:
You will need to read the cookie as well. It is argument for the CustomMarshaler GetInstance method.